src/components/PupilViews/PupilReview/PupilReview.view.tsx

```diff
@@ -144,13 +146,43 @@ export const PupilViewsReview = (props: PupilViewsReviewProps) => {
       })}`;
       navigator.clipboard.writeText(shareUrl);
       setIsAttemptingShare("shared");
+      if (sectionResults["starter-quiz"]?.isComplete) {
+        track.activityResultsSharedStarterQuiz({
+          shareMedium: "copy-link",
+          pupilQuizGrade: sectionResults["starter-quiz"]?.grade ?? 0,
+          pupilQuizNumQuestions:
+            sectionResults["starter-quiz"]?.numQuestions ?? 0,
+        });
+      }
+      if (sectionResults["exit-quiz"]?.isComplete) {
+        track.activityResultsSharedExitQuiz({
+          shareMedium: "copy-link",
+          pupilQuizGrade: sectionResults["exit-quiz"]?.grade ?? 0,
+          pupilQuizNumQuestions: sectionResults["exit-quiz"]?.numQuestions ?? 0,
+        });
+      }
     }
   };
 
   if (phase === "foundation") {
     throw new Error("Foundation phase is not supported");
   }
 
+  track.lessonSummaryReviewed({
```

Comment:
this will call every rerender ! You need to put as conditional around it with state to catch it only on the first render.